### ISSUE TITLE
Obvious errors with coroutines are not detected

### ISSUE THREAD

**Issue by WhyNotHugo** [python/mypy#10841]
**Bug Report**

Some obvious type issues with `coroutines` are not detected. See two examples below.

**To Reproduce**

```python
import asyncio


async def function_one():
    return 1


async def function_two():
    function_one()[1]
    function_one() + 1


asyncio.run(function_two())
```

**Expected Behavior**

`mypy` should detect two `TypeError` here:

- `'coroutine' object is not subscriptable`
- `unsupported operand type(s) for +: 'coroutine' and 'int'`

**Actual Behavior**

`Success: no issues found in 1 source file`

**Your Environment**

mypy 0.910
Python 3.9.6

**Comment by hauntsaninja** [comment]
https://mypy.readthedocs.io/en/stable/common_issues.html#no-errors-reported-for-obviously-wrong-code

**Comment by WhyNotHugo** [comment]
Odd, a couroutine is _never_ indexable, shouldn't mypy be able to detect this even if it's not type-annotated?

**Comment by JelleZijlstra** [comment]
It can, yes, but mypy simply never provides errors for functions that aren't type checked.

**Comment by JelleZijlstra** [comment]
You can use the `--check-untyped-defs` option to change this behavior.
